OpenRA.Game/Traits/Target.cs

```diff
@@ -173,6 +163,48 @@ public bool IsInRange(WPos origin, WDist range)
 			return Positions.Any(t => (t - origin).HorizontalLengthSquared <= range.LengthSquared);
 		}
 
+		public bool IsOwnerTargetable(Actor self)
+		{
+			if (this.FrozenActor != null)
+				return this.FrozenActor.Visible;
+			else if (this.Actor != null)
+				return self.Owner.CanTargetActor(this.Actor);
+
+			return true;
+		}
+
+		public Target TryUpdateFrozenActorTarget(Actor self)
```

Comment:
I suggest changing this to `public Target UpdateForAttacker(Actor attacker)` and then changing the default case at the end from `return Invalid` to `return this`.

In the short term this simplifies the way we handle the frozen &lrarr; actor switch, which i'll comment about in `ExternalCapture`.

In the long term (which we shouldn't worry about here) it could be relatively easily extended to switch targets when an actor transforms into something else (e.g. MCV &larr; Construction Yard).